### PR TITLE
[JBTM-3364] LRA recovery url creation to consider deployment path

### PR DESCRIPTION
**Pull request by ochaloup**
https://issues.redhat.com/browse/JBTM-3364

LRA
!MAIN !CORE !QA_JTA !QA_JTS_JDKORB !QA_JTS_OPENJDKORB !QA_JTS_JACORB !BLACKTIE !XTS !PERF NO_WIN !RTS !AS_TESTS !TOMCAT !JACOCO

### CODE REVIEW

**Review by xstefank — APPROVED** [review]

File: rts/lra/lra-coordinator-jar/src/main/java/io/narayana/lra/coordinator/domain/service/LRAService.java [hunk 1]
```diff
@@ -57,6 +57,8 @@
 
 @ApplicationScoped
 public class LRAService {
+    private static final Pattern LINK_REL_PATTERN = Pattern.compile("(\\w+)=\"([^\"]+)\"|([^\\s]+)");
```
Comment (mmusgrov) [inline rts/lra/lra-coordinator-jar/src/main/java/io/narayana/lra/coordinator/domain/service/LRAService.java:60]: This change seems unrelated. Let's keep PR limited to the issue that is being fixed.

**Review by mmusgrov — CHANGES_REQUESTED** [review]

File: rts/lra/lra-coordinator-jar/src/main/java/io/narayana/lra/coordinator/domain/service/LRAService.java (hunk 1 above)
Comment (mmusgrov) [reply rts/lra/lra-coordinator-jar/src/main/java/io/narayana/lra/coordinator/domain/service/LRAService.java:60]: Ah wait, this is a fixing a performance problem since the code is compiling the pattern on every call so your change is needed, sorry.

**Review by mmusgrov — APPROVED** [review]

File: rts/lra/lra-coordinator-jar/src/main/java/io/narayana/lra/coordinator/domain/service/LRAService.java (hunk 1 above)
Comment (ochaloup) [reply rts/lra/lra-coordinator-jar/src/main/java/io/narayana/lra/coordinator/domain/service/LRAService.java:60]: yes. ok, thanks.
